### PR TITLE
Update parameter requirement for manifest-commit-lock workflow

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
Remove the mandatory RC number field for which is not required for UPDATE_TO_RECENT_COMMITS option.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/release-workflows/release-manifest-commit-lock.jenkinsfile [hunk 1]
```diff
@@ -60,7 +60,7 @@ pipeline {
                                 currentBuild.result = 'ABORTED'
                                 error('MANIFEST_LOCK_ACTION and/or RELEASE_VERSION cannot be empty!')
                     }
-                    if ((params.MANIFEST_LOCK_ACTION == 'MATCH_BUILD_MANIFEST' || params.MANIFEST_LOCK_ACTION == 'UPDATE_TO_RECENT_COMMITS') && (OPENSEARCH_RELEASE_CANDIDATE.isEmpty() || OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE.isEmpty())) {
+                    if (params.MANIFEST_LOCK_ACTION == 'MATCH_BUILD_MANIFEST' && (OPENSEARCH_RELEASE_CANDIDATE.isEmpty() || OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE.isEmpty())) {
                                 currentBuild.result = 'ABORTED'
                                 error('OPENSEARCH_RELEASE_CANDIDATE and/or OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE cannot be empty when MANIFEST_LOCK_ACTION is MATCH_BUILD_MANIFEST or UPDATE_TO_RECENT_COMMITS.')
```
Comment (gaiksaya) [inline jenkins/release-workflows/release-manifest-commit-lock.jenkinsfile:65]:
```suggestion
                                error('OPENSEARCH_RELEASE_CANDIDATE and/or OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE cannot be empty when MANIFEST_LOCK_ACTION is MATCH_BUILD_MANIFEST.')
```

File: tests/jenkins/TestReleaseManifestCommitLock.groovy [hunk 2]
```diff
@@ -49,13 +50,27 @@ class TestReleaseManifestCommitLock extends BuildPipelineTest {
...
+    @Test
+    public void testManifestCommitLock_matchBuildManifest_exception() {
+        addParam('MANIFEST_LOCK_ACTION', 'MATCH_BUILD_MANIFEST')
+        binding.setVariable('OPENSEARCH_RELEASE_CANDIDATE', '')
+        binding.setVariable('OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE', '')
+        Exception exception = assertThrows(Exception) {
+        runScript('jenkins/release-workflows/release-manifest-commit-lock.jenkinsfile')
+        }
+        def callStack = helper.getCallStack()
+        assertCallStack().contains('OPENSEARCH_RELEASE_CANDIDATE and/or OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE cannot be empty when MANIFEST_LOCK_ACTION is MATCH_BUILD_MANIFEST or UPDATE_TO_RECENT_COMMITS.')
```
Comment (gaiksaya) [inline tests/jenkins/TestReleaseManifestCommitLock.groovy:71]:
```suggestion
        assertCallStack().contains('OPENSEARCH_RELEASE_CANDIDATE and/or OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE cannot be empty when MANIFEST_LOCK_ACTION is MATCH_BUILD_MANIFEST.')
```

Comment (Divyaasm) [reply tests/jenkins/TestReleaseManifestCommitLock.groovy]: Done @gaiksaya

**Review by gaiksaya — APPROVED** [review]

File: tests/jenkins/TestReleaseManifestCommitLock.groovy (hunk 2 above)
Comment (Divyaasm) [reply tests/jenkins/TestReleaseManifestCommitLock.groovy]: Thanks! will merge after the release

**Pull request merged**